src/mpi/romio/adio/ad_gpfs/ad_gpfs_hints.c

```diff
@@ -106,6 +103,11 @@ void ADIOI_GPFS_SetInfo(ADIO_File fd, MPI_Info users_info, int *error_code)
             ADIOI_Free(fd->hints->cb_config_list);
         fd->hints->cb_config_list = NULL;
 
+        if (value == NULL) {
```

Comment:
Is there a noticeable issue that this code solves? I'm tempted to drop it because it's a lot of churn and duplication.

Reply:
It reduces memory footprint. We encountered the memory space problem from an ECP application.